internal/ingest/ingest.go

```diff
@@ -276,11 +276,12 @@ func (ing *Ingester) Sync(ctx context.Context, peerID peer.ID, peerAddr multiadd
 		// selector is nil.
 		var seenAdCids []cid.Cid
 		opts := []legs.SyncOption{
+			legs.AlwaysUpdateLatest(),
 			legs.ScopedBlockHook(func(i peer.ID, c cid.Cid) {
 				seenAdCids = append(seenAdCids, c)
 			}),
 		}
-		if sel != nil && !resync {
+		if sel != nil && !resync && latest != cid.Undef {
 			opts = append(opts, legs.CheckAlreadySynced())
```

Comment:
with https://github.com/filecoin-project/go-legs/pull/97 I think we could get rid of this block

Reply:
exterminated